internal/runner/operation_project_destroy.go

```diff
@@ -0,0 +1,71 @@
+package runner
+
+import (
+	"context"
+	"github.com/hashicorp/waypoint/internal/core"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+func (r *Runner) executeDestroyProjectOp(
+	ctx context.Context,
+	job *pb.Job,
+	project *core.Project,
+) (*pb.Job_Result, error) {
+	client := project.Client()
+	destroyProjectOp, ok := job.Operation.(*pb.Job_DestroyProject)
+	if !ok {
+		// this shouldn't happen since the call to this function is gated
+		// on the above type match.
+		panic("operation not expected type")
+	}
+
+	// Update the project state to indicate that it's being destroyed
+	// TODO: Do we need to do something with the response here?
+	_, err := client.UpsertProject(ctx,
+		&pb.UpsertProjectRequest{
+			Project: &pb.Project{
+				Name:  destroyProjectOp.DestroyProject.Project.Name,
+				State: pb.Project_DESTROYING,
+			},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	if !destroyProjectOp.DestroyProject.SkipDestroyResources {
+		for _, app := range project.Apps() {
+			// TODO: Do we need to do something with the response here?
+			_, err := r.executeDestroyOp(ctx,
```

Comment:
I don't think so, looks like this returns an empty `pb.Job_Result{}`

https://github.com/hashicorp/waypoint/blob/25b2130a43e4594e67c897884f6934aa0f19d9b9/internal/runner/operation_destroy.go#L42